docs/cli/ilab-config-current-state.md

```diff
@@ -0,0 +1,90 @@
+
+- **Click Extensions** `clickext.py` defines some helpful extensions to click functionality that we use throughout the config process. Specifically, `clickext` manages the funnelling of `config` defaults into `click` options. The `clickext.ConfigOptions` class maps values stored in `ctx.obj.config` and `ctx.obj.default_map` to the click arguments defined in a subcommand decorator, for example:
+
+```python
+@click.option(
+    "--data-path",
+    ...
+    cls=clickext.ConfigOption,
+    typ
+)
+```
+
+This allows the function to read the values stored in the context object to fill that variable.
+
+### Additional Args
+
+A subset of the `ilab model train` config variables were deemed more advanced and less likely to be tweaked, so they are exported to a separate section of the config names "additional_args". For example, on linux, the training additional args are stored in `~/.local/share/instructlab/internal/train_configuration/additional/additional_args.yaml`.
+
+For these, click needs separate functionality to be able to map any user-provided args to the config section. We enable this by specifying `config_sections` in the respective decorators.
+
+```python
+@click.option(
+    "--learning-rate",
+    ...
+    cls=clickext.ConfigOption,
+    config_sections="additional_arguments",
+    ...
+```
+
+### Global Config
+
+By default, `ilab` looks for and saves the config in `~/Library/Application Support/instructlab/config.yaml` on mac and `~/.config/instructlab/config.yaml` on Linux. Optionally, users can write their own config.yaml and set it's path to the `ILAB_GLOBAL_CONFIG` environment variable. Upon running `ilab config init`, the file located at that path will be copied and written to the default config path expected by `ilab`.
+
+## Config Flow for `ilab` Commands
+
+1. When the user runs an ilab command, click captures the CLI args provided by the user and then the flow is passed to `init()` in `configuration.py`.
+1. `init()` determines which config file to use and passes the flow to `read_config`, which will read the config file and return a `Config` object back to `init()`.
+1. `init()` takes the `Config` object and assigns it to 2 different fields under the click context :
+    1. generates a new `Lab` object and assigns it to the click context's `obj` field (`ctx.obj = Lab(config_obj, config_file, error_msg)`)
+    1. parses it into a dict and assigns it to `ctx.default_map`
+1. `init()` finally checks to see if any of the args provided by the user were 'additional args' for training, and creates a dict with the defaults filled in. This dict is assigned to `ctx.default_map["train"]["additional_args"]`
+1. The flow of control is then passed to the subcommand that was called (e.g. generate, train, evaluate). The command group (e.g. data, model) inherits the context from `ilab` so that the subcommand can access the fields under `ctx` by accepting it as a function argument. For example:
```

Comment:
IMO this section should probably also mention the `click.core.ParameterSource` check that determines what the source of a particular field value is, since this affects what the eventual config obj in memory ends up looking like when the user runs `ilab config init` 
https://github.com/instructlab/instructlab/blob/main/src/instructlab/config/init.py#L145
maybe after this point somewhere